Update changelog and authors for unyt 2.9.0 (#246)
yt-project/unyt: HISTORY.rst

History
=======

2.9.0 (2022-07-13)
------------------

> * Dropped support for Python 3.6 and 3.7.

Reviewer: ```suggestion
* Dropped support for Python 3.5, 3.6 and 3.7.
```

Reviewer: I see that 3.5 support was "dropped" already in version 2.8, according to the changelog, however I confirm that Python 3.5 was still documented as supported according to Python classifiers and `python_requires`. See 
https://github.com/yt-project/unyt/blob/eeefa00a2fddbf0dba6ab854e968ef43e31f851e/setup.py#L51

Author: To reduce confusion, I'm going to leave this line as-is.

Reviewer: Fine by me

> * Added support for Python 3.9 and 3.10.

Reviewer: ```suggestion
* Added support for Python 3.8, 3.9 and 3.10.
```

  values in conversions. See `PR #176
  <https://github.com/yt-project/unyt/pull/176>`_. Thank you to Lee Johnston
  (@l-johnston on GitHub) for the contribution.
* Fix an issue when using ``matplotlib_support``, plot an empty unyt array,
  would result in an error when changing units. See `PR #180
  <https://github.com/yt-project/unyt/pull/180>`_. Thank you to Josh Borrow
  (@JBorrow on GitHub) for the contribution.
* Fix an issue where units would be printed twice in formatted strings with
  an ``unyt_array`` embedded. See `PR #188
  <https://github.com/yt-project/unyt/pull/180>`_. Thank you to Clément Robert
  (@neutrinoceros on GitHub) for the contribution.
* Add a method to parse a ``unyt_quantity`` from a string expression. See `PR #191
  <https://github.com/yt-project/unyt/pull/191>`_. Thank you to Clément Robert
  (@neutrinoceros on GitHub) for the contribution.
* Fix an issue where a ``unyt_array`` with dtype int8 could not be converted
  to a different unit. See `PR #197
  <https://github.com/yt-project/unyt/pull/197>`_. Thank you to Clément Robert
  (@neutrinoceros on GitHub) for the contribution.

> * The import time for ``unyt`` has been optimized. See `PR #199

Reviewer: Modesty++
Also this particular PR should probably be mentioned for the actual change that's performed. It's still (remotely) possible that it's breaking to someone.
```suggestion
* The import time for ``unyt`` has been reduced. See `PR #199
```

  (@JBorrow on GitHub) for the contribution.
* Add support for coercing iterables of ``unyt_array`` objects with nonuniform
  dimensionally equivalent units to a single ``unyt_array``. See `PR #211
  <https://github.com/yt-project/unyt/pull/211>`_. Thank you to Nathan Goldbaum
  (@ngoldbaum on GitHub) for the contribution.
* Add the civil engineering units ``pli``, ``plf``, ``psf``, ``kli``, ``klf``,
  and ``ksf``. See `PR #217 <https://github.com/yt-project/unyt/pull/217>`_.
  Thank you to @osnippet on GitHub for the contribution.
* Fix typos in constants and unit prefixes. See `PR #218
  <https://github.com/yt-project/unyt/pull/218>`_. Thank you to Clément Robert
  (@neutrinoceros on GitHub) for the contribution.
* Fix an issue where multiplying a 1-element ``unyt_array`` would return a
  ``unyt_quantity``. See `PR #225 <https://github.com/yt-project/unyt/pull/225>`_.
  Thank you to Clément Robert (@neutrinoceros on GitHub) for the contribution.
* Add the Rydberg constant ``R_∞`` and unit ``Ry``, add the dimension
  ``angular_frequency`` and the unit ``rpm``, and increase the precision of
  Avogadro's number. See `PR #228 <https://github.com/yt-project/unyt/pull/228>`_.

> * Fix an issue where ``np.divide.reduce`` returns incorrect units for ``unyt_array``

Reviewer: for tense consistency
```suggestion
* Fix an issue where ``np.divide.reduce`` would return incorrect units for ``unyt_array``
```

  instances. See `PR #230 <https://github.com/yt-project/unyt/pull/230>`_.
  Thank you to Kyle Oman (@kyleaoman on GitHub) for the contribution.
* Fix a regression where ``unyt_array.reshape`` wasn't wrapping
  ``numpy.ndarray.reshape`` correctly. See `PR #242
  <https://github.com/yt-project/unyt/pull/242>`_. Thank you to Clément Robert

>   (@neutrinoceros on GitHub) for the contribution.

Reviewer: Not sure this is worth mentioning in the changelog: this regression was never released, so it's not a diff wrt to 2.8.0

Author: If it's ok with you, we can just remove it then
